### ISSUE TITLE
alpha-25008,c8d96591 De sables all add-ons

### ISSUE THREAD

**Issue by tapper82** [nvaccess/nvda#13502]
Hi just installed alpha-25008,c8d96591. All my add-ons were disabled. I reverted back to the older build and all works as it should. Working good build is alpha-25000,8220d6c3

**Comment by seanbudd** [comment]
Yes, NVDA 2022.1 is an addon compatibility breaking release. Addons may already working with your alpha build, consider using 2021.3.4 until addons are updated.

**Comment by tapper82** [comment]
It would of bin nice to get a update about this! May be some one should post a heads up on the twitter.

**Comment by CyrilleB79** [comment]
@seanbudd, @feerrenrut:
Now that alpha and beta branches have back compa API set to 2022.1, can we consider that 2022.1 add-on API is freezed.

If yes, NVAccess should consider the following actions:
1. Send a message to add-on mailing list to inform of the translation freeze; an alternative action is of course to release NVDA 2022.1beta1 today (but maybe you may want to delay it for a few days to avoid confusion with the just released 2021.3.4)
2. Accept the add-ons tagged 2022.1 compatible on addonFile repo, unless you prefer to have a second confirmation from add-on authors who had submitted their add-ons before the API freeze. Cc @nvdaes, @josephsl

**Comment by feerrenrut** [comment]
@tapper82 the first release of the year (`20XY.1`) Should always be expected as a compatibility breaking release. This gives NVDA core developers an opportunity to rectify issues holding back development.

We have tried to communicate this:
- Opened a PR at the start of the dev cycle (Oct 18, 2021) for the API version change:
[update add-on api back compat for 2022.1 by feerrenrut · Pull Request #12951 · nvaccess/nvda (github.com)](https://github.com/nvaccess/nvda/pull/12951)
- Included a note in the changes file (since Oct 18, 2021) that 2022.1 will be a compatibility breaking release. [Start dev cycle 2022.1 by feerrenrut · Pull Request #12949 · nvaccess/nvda (github.com)](https://github.com/nvaccess/nvda/pull/12949)

Be aware, alpha builds are not considered stable, please don't rely on them. When using alpha builds you are acting as a tester for the project, this is appreciated, but if you want stability please use the stable releases.

**Comment by feerrenrut** [comment]
Thanks @CyrilleB79, please note we do have a process in place here.
> can we consider that 2022.1 add-on API is freezed.

Yes, that is correct.

The addons not yet merged on addonFiles need review or updates. Addon authors are expected to test their addon with the beta when it becomes available.

> add-on mailing list to inform of the translation freeze

Translation freeze comes later and isn't related to add-ons.

When the first beta is made the add-on API is considered set unless there are exceptional circumstances. The beta period is the final opportunity for changes based on conversation between add-on authors and core developers.
Add-on authors should use this time to confirm the changes to the API will not prevent them from updating their add-on.
It's our signal that we think we are done with the changes to the API.

**Comment by tapper82** [comment]
I know about the stability  of test builds. It still would of bin nice to have some one post to twitter to say that the next build will brake all add-ons. I test the testing builds, but I don't keep track of all pulls on Github never mind one from as far back as oct 18th 2021

**Comment by josephsl** [comment]
Hi all,

I can assure everyone that I will release versions of my add-ons compatible with NVDA 2022.1 as soon as beta 1 is released, beginning with Add-on Updater (internally, my add-ons are being prepared; I'm waiting for a specific message to show up in Git before taking the next step; see below, and read an overview posted by Reef above my notes). The reason for the delay is the fact that, as of time of this comment, beta 1 is not tagged.

For reference (for people who will come across this and similar issues in the future, as well as a note to self), the steps I take to get an add-on ready for future NVDA releases (including backwards incompatible release) are:

1. Live testing: as some of you may know by now, I test add-ons against alpha and beta releases and keep a copy of a stable version for stable build testing. As Reef pointed out, alpha is bleeding-edge, so things like the issue reported will be a common thing (I'm paraphrasing NV Access documentation here).
2. Branch exchanges in NVDA Core: a few weeks prior to beta 1 releases, a "branch exchange" takes place where code from master (alpha) branch is merged into beta branch. This is done so translators can get started on evaluating and translating user interface messages for the upcoming release. Then once beta 1 is released (or shortly before), updated data from beta branch is merged into master branch. This is the situation described in this case (latest backward compatibility flag was merged into beta, and now it was merged into master (alpha) branch). I take branch exchange as a point where I need to perform additional add-on testing to make sure add-ons are getting ready for the upcoming beta cycle (I won't say this is the final testing phase because things can change between branch exchange and beta 1 release).
3. Tags: NVDA releases meant for wider distribution are tagged using Git, including betas, RC's, and stable releases. The message I (and I assume others) am looking for is a tag creation message (tags are of the form release-version(none/betaN/rcN) where none signifies a stable release, and betaN and rcN refer to respective beta and release candidate milestones). Therefore the specific message I refer to is a Git message telling people that a tag named "release-2022.1beta1" (without quotes) was created and pushed. Only then I and other add-on authors are assured that it is safe to submit new add-on releases declaring support for the upcoming release (and for folks to perform add-on testing if needed) as the API is frozen with the beta 1 tag.
4. Last minute add-on checks: just because API is frozen does not mean all add-ons are ready. There may have been last minute changes that could break add-ons, and it is important to test add-on installation and compatibility steps. For my add-ons, only after I test installation steps with the packages I plan to submit would I deem add-ons ready for the upcoming NVDA release and submit the updated add-on packages.

In case this is a year.1 release of NVDA, I take extra precautions because things can be change and removed (of course changes for developers section in the what's new document notes things that will be changing and/or removed in the next year.1 release). But as far as add-on compatibility release scheduling goes, the above is the procedure I follow based on recommendations from NV Access (API's can change up until beta 1 is released, and I release compatibility updates as soon as beta 1 is tagged - sometimes within minutes, but usually within hours).

Thanks.

**Comment by tapper82** [comment]
@josephsl Thanks for the update mate. Plus thanks for all your work on add-ons.

**Comment by CyrilleB79** [comment]
@josephsl, actually, I had CCed you as well as @nvdaes only because you are the reviewers of the PRs in addonFile repo before NVAccess' review. I have not been clear about it on my first post, sorry.
Thanks for the detailed explanation regarding your add-ons anyway!

**Comment by Brian1Gaff** [comment]
Maybe make a post in the  lists as well. Not everyone is on Twitter or has 
the time!
 Brian

***@***.***
Sent via blueyonder.
Please address personal E-mail to:-
***@***.***, putting 'Brian Gaff'
in the display name field.
Newsgroup monitored: alt.comp.blind-users
----- Original Message -----

**Comment by Brian1Gaff** [comment]
However the casual user would not see any of those. It would need some 
explicit obvious non programmer info to be put up, I feel. When they say 
interface breaking, most of the time its just the odd add on that suffers, 
but this one is most of them. A bit of a bigger issue, I'd suggest.
 Brian

***@***.***
Sent via blueyonder.
Please address personal E-mail to:-
***@***.***, putting 'Brian Gaff'
in the display name field.
Newsgroup monitored: alt.comp.blind-users
----- Original Message -----

**Comment by Brian1Gaff** [comment]
I agree, but as I say, twitter is not on ly daily list of things to do, but 
reading these groups is.
 I think the end user also needs to be warned when updating to it when it 
becomes stable, before they burn their bridges.
 Brian

***@***.***
Sent via blueyonder.
Please address personal E-mail to:-
***@***.***, putting 'Brian Gaff'
in the display name field.
Newsgroup monitored: alt.comp.blind-users
----- Original Message -----

**Comment by Brian1Gaff** [comment]
I still say that as not all add on writers are always omnipresent, with such 
a major add on breaker as this, some extra work might be required to get the 
writers alerted to the changes.
 Brian

***@***.***
Sent via blueyonder.
Please address personal E-mail to:-
***@***.***, putting 'Brian Gaff'
in the display name field.
Newsgroup monitored: alt.comp.blind-users
----- Original Message -----
